wolfcrypt/src/chacha20_poly1305.c

```diff
@@ -293,4 +293,158 @@ int wc_ChaCha20Poly1305_Final(ChaChaPoly_Aead* aead,
     return ret;
 }
 
+#ifdef HAVE_XCHACHA
+
+int wc_XChaCha20Poly1305_Init(
+    ChaChaPoly_Aead *aead,
+    const byte *ad, word32 ad_len,
+    const byte *nonce, word32 nonce_len,
+    const byte *key, word32 key_len,
+    int isEncrypt)
+{
+    byte authKey[CHACHA20_POLY1305_AEAD_KEYSIZE];
+    int ret;
+
+    if ((ad == NULL) || (nonce == NULL) || (key == NULL))
+        return BAD_FUNC_ARG;
+
+    if ((key_len != CHACHA20_POLY1305_AEAD_KEYSIZE) ||
+        (nonce_len < XCHACHA20_POLY1305_AEAD_NONCE_SIZE))
```

Comment:
I believe XChaCha20-Poly1305 is formally only defined for a nonce of 24 bytes. I think it might be better to reject `nonce_len` if it is not exactly `XCHACHA20_POLY1305_AEAD_NONCE_SIZE`.

Reply:
I agree.  Done.